### ISSUE TITLE
better support for grep/searching contents of multiple files/directories

### ISSUE THREAD

**Issue by raycewest** [ranger/ranger#1243]
Don't see this being addressed anywhere - ranger currently supports the grep command, which AFAIK only works on marked files (takes time), outputs to separate less-like buffer (ranger can't further interact with the results), and doesn't seem to take it's normal command-line options.  

What're everyone's thoughts on a command that recursively greps (e.g., 'grep -rn "." -e "string") the current directory for text?  Anyone already have an elegant solution to this?  I feel like it's a typical use-case.

**Comment by toonn** [comment]
I might implement a custom command for this. What did you have in mind for integration though? Mark all the files that contain a match?

**Comment by raycewest** [comment]
Not 100% sure what would make most sense, but next/prev would probably be
desired.  Marking would make sense in some cases.

**Comment by toonn** [comment]
What do you expect from next/prev that you can't do in the pager rn?

**Comment by raycewest** [comment]
What I mean is, if say all of the resulting directories/files are marked as
a result of this search, then I would think next/prev should traverse only
those, so you can peruse the search results just as you would if you were
just searching names.

**Comment by toonn** [comment]
So, not mark the files but move between them, like search does? Displaying the grep matches in the preview column or something?

**Comment by raycewest** [comment]
Yeah, that might be a good default.  Marking or other effects could be
desirable as options.

**Comment by frankgerhardt** [comment]
Does anybody have a workaround?

**Comment by chu4ng** [comment]
I just bind this script to `<c+f>`:
```shell
#!/bin/sh
if [[ "$1" -eq "-d" ]]; then path_="$2"; shift 2; fi 
place=$(rg -j 50 --color 'ansi' -n --smart-case -e "$*" $path_ | fzf --ansi -1 | awk -F':' -e '{print $1" +"$2}')
if [[ -n "$place" ]]; then ( $TERM -e $SHELL -c "$EDITOR $place" &>/dev/null & ); fi
```
I'm not sure about `-d` flag, because ranger can translate `%s` to `'file1' 'file2'` or to `file1 file2`, so we need change logic or whatever. May be add 'tab' function for this macros to translate '%s' to actually selected files inside console?

I think it's the best solution with current ranger functionality. We can rewrite it as module, but it will be repeat fzf functionality.
